### PR TITLE
ESP-IDF Firmware Update Example

### PR DESCRIPTION
**Pull request by rashmivaidya72**

### CODE REVIEW

**Review by sam-golioth — COMMENTED** [review]
Changes look good, but tests are failing

File: examples/esp_idf/README.md [hunk 1]
```diff
@@ -77,5 +77,9 @@ and deletes it after the test ends.
 The test can be invoked by the following command from the <sample>/pytest location:
 
 ```sh
-pytest -s --rootdir . --port /path/to/serial/port --board esp-idf --wifi-ssid your-wifi-ssid --wifi-psk your-wifi-psk --api-key your-api-key --fw-image /path/to/merged-binary
+pytest -s --rootdir . --port /path/to/serial/port --board your-esp-board-name --wifi-ssid your-wifi-ssid --wifi-psk your-wifi-psk --api-key your-api-key --fw-image /path/to/merged-binary
 ```
+
+For esp32c3, use --board esp32c3_devkitm_espidf
+For esp32s3, use --board esp32s3_devkitc_espidf
+For esp32 wrover, use --board esp32_devkitc_wrover
```
Comment (sam-golioth) [inline examples/esp_idf/README.md:85]:
```suggestion
For esp32 wrover, use --board esp32_devkitc_wrover_espidf
```

File: examples/esp_idf/fw_update/main/app_main.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    // Initialize NVS first.
+    nvs_init();
+
+    // Create a background shell/CLI task (type "help" to see a list of supported commands)
+    shell_start();
+
+    // If the credentials haven't been set in NVS, we will wait here for the user
+    // to input them via the shell.
+    if (!nvs_credentials_are_set())
```
Comment (sam-golioth) [inline examples/esp_idf/fw_update/main/app_main.c:45]: We should use the new sample credentials module here

File: examples/esp_idf/fw_update/sdkconfig.defaults [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+CONFIG_PARTITION_TABLE_CUSTOM=y
+CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions.csv"
+
+# If the new app firmware does not cancel the rollback,
+# then this will ensure that rollback happens automatically
+# if/when the device is next rebooted.
+CONFIG_BOOTLOADER_APP_ROLLBACK_ENABLE=y
+
+# Golioth Services
+CONFIG_GOLIOTH_RPC=y
```
Comment (sam-golioth) [inline examples/esp_idf/fw_update/sdkconfig.defaults:28]: Do we need RPC?

File: examples/esp_idf/fw_update/main/app_main.c (hunk 2 above)
Comment (rashmivaidya72) [reply examples/esp_idf/fw_update/main/app_main.c:45]: Right!

File: examples/esp_idf/fw_update/sdkconfig.defaults (hunk 3 above)
Comment (rashmivaidya72) [reply examples/esp_idf/fw_update/sdkconfig.defaults]: Nope. Removing.

**Review by sam-golioth — APPROVED** [review]

**Pull request merged**